### PR TITLE
Decompile main menu

### PR DESCRIPTION
**Pull request by golemgalvanize**

### CODE REVIEW

File: src/main_menu.c [hunk 1]
```diff
@@ -1,15 +1,1955 @@
...
+{
+    if (!sub_8197224())
+        gTasks[taskId].func = task_new_game_prof_birch_speech_16;
+}
+
+void task_new_game_prof_birch_speech_16(u8 taskId)
+{
+    if ((gMain.newKeys & A_BUTTON) || (gMain.newKeys & B_BUTTON))
+    {
+        BeginNormalPaletteFade(-1, 0, 0, 16, 0);
```
Comment (Diegoisawesome) [inline src/main_menu.c]: RGB_BLACK

File: src/main_menu.c [hunk 2]
```diff
@@ -1,15 +1,1955 @@
...
+    u8 spriteId;
+    
+    if (gTasks[taskId].data[5])
+    {
+        gSprites[gTasks[taskId].data[10]].invisible = 1;
+        gSprites[gTasks[taskId].data[11]].invisible = 1;
+        spriteId = gTasks[taskId].data[8];
+        gSprites[spriteId].pos1.x = 0x88;
+        gSprites[spriteId].pos1.y = 0x3C;
+        gSprites[spriteId].invisible = 0;
```
Comment (Diegoisawesome) [inline src/main_menu.c]: FALSE

File: src/main_menu.c [hunk 3]
```diff
@@ -1,15 +1,1955 @@
...
+    }
+}
+
+void task_new_game_prof_birch_speech_part2_6(u8 taskId)
+{
+    u8 spriteId;
+    
+    if (gTasks[taskId].data[5])
+    {
+        gSprites[gTasks[taskId].data[10]].invisible = 1;
```
Comment (Diegoisawesome) [inline src/main_menu.c]: TRUE

File: src/main_menu.c [hunk 4]
```diff
@@ -1,15 +1,1955 @@
...
+        gSprites[gTasks[taskId].data[2]].oam.objMode = 0;
+        if (!sub_8197224())
+        {
+            spriteId = gTasks[taskId].data[2];
+            gSprites[spriteId].oam.affineMode = 1;
+            gSprites[spriteId].affineAnims = &gUnknown_082FF114;
+            InitSpriteAffineAnim(&gSprites[spriteId]);
+            StartSpriteAffineAnim(&gSprites[spriteId], 0);
+            gSprites[spriteId].callback = sub_80318D8;
+            BeginNormalPaletteFade(0xFFFF, 0, 0, 16, 0);
```
Comment (Diegoisawesome) [inline src/main_menu.c]: RGB_BLACK, and make the 0xFFFF be 0x0000FFFF

File: src/main_menu.c [hunk 5]
```diff
@@ -1,15 +1,1955 @@
...
+void task_new_game_prof_birch_speech_part2_11(u8 taskId)
+{
+    u8 spriteId;
+    
+    if (!gPaletteFade.active)
+    {
+        spriteId = gTasks[taskId].data[2];
+        gSprites[spriteId].callback = nullsub_11;
+        SetGpuReg(REG_OFFSET_DISPCNT, DISPCNT_OBJ_ON | DISPCNT_OBJ_1D_MAP);
+        BeginNormalPaletteFade(-0x10000, 0, 0, 16, -1);
```
Comment (Diegoisawesome) [inline src/main_menu.c]: RGB_WHITE2

File: src/main_menu.c [hunk 6]
```diff
@@ -1,15 +1,1955 @@
...
+void AddBirchSpeechObjects(u8 taskId)
+{
+    u8 spriteId = AddNewGameBirchObject(0x88, 0x3C, 1);
+    u8 spriteId2;
+    u8 spriteId3;
+    u8 spriteId4;
+    
+    gSprites[spriteId].callback = nullsub_11;
+    gSprites[spriteId].oam.priority = 0;
+    gSprites[spriteId].invisible = 1;
```
Comment (Diegoisawesome) [inline src/main_menu.c]: Invisible should be TRUE or FALSE

File: src/main_menu.c [hunk 7]
```diff
@@ -1,15 +1,1955 @@
...
+    u16 r9 = 1 + baseTileNum;
+    u16 r10 = 2 + baseTileNum;
+    u16 sp18 = 3 + baseTileNum;
+    u16 spC = 5 + baseTileNum;
+    u16 sp10 = 6 + baseTileNum;
+    u16 sp14 = 7 + baseTileNum;
+    u16 r6 = 8 + baseTileNum;
+    
+    FillBgTilemapBufferRect(template->priority, baseTileNum, template->tilemapLeft - 1, template->tilemapTop - 1, 1, 1, 2);
+    FillBgTilemapBufferRect(template->priority, r9, template->tilemapLeft, template->tilemapTop - 1, template->width, 1, 2);
```
Comment (Diegoisawesome) [inline src/main_menu.c]: These temp vars are absolutely necessary?

File: src/main_menu.c [hunk 8]
```diff
@@ -1,15 +1,1955 @@
...
+
+void sub_8032298(u8 a, u8 b, u8 c, u8 d, u8 e, u8 unused)
+{
+    FillBgTilemapBufferRect(a, 0, b + 0xFF, c + 0xFF, d + 2, e + 2, 2);
+}
+
+void sub_80322E0(u8 windowId, u8 a)
+{
+    CallWindowFunction(windowId, sub_8032298);
+    FillWindowPixelBuffer(windowId, 17);
```
Comment (Diegoisawesome) [inline src/main_menu.c]: Make that 0x11

File: src/main_menu.c [hunk 9]
```diff
@@ -1,15 +1,1955 @@
...
+    }
+}
+
+void sub_80323CC(u8 a, u8 b, u16 c, u16 d, u8 e, u8 f)
+{
+    struct WindowTemplate sp;
+    struct WindowTemplate sp2;
+    
+    sp = sub_8198A50(0, a + 1, b + 1, 5, 4, f, d);
+    sp2 = sp;
```
Comment (Diegoisawesome) [inline src/main_menu.c]: This probably isn't needed and is just a result of the weird return-by-value nature of the function

File: src/main_menu.c [hunk 10]
```diff
@@ -1,15 +1,1955 @@
...
+    FillBgTilemapBufferRect(a, 0x101, b+d-1, c-1, 1,   1, f);
+    FillBgTilemapBufferRect(a, 0x102, b+d,   c-1, 1,   1, f);
+    FillBgTilemapBufferRect(a, 0x103, b-2,   c,   1,   5, f);
+    FillBgTilemapBufferRect(a, 0x105, b-1,   c,   d+1, 5, f);
+    FillBgTilemapBufferRect(a, 0x106, b+d,   c,   1,   5, f);
+    FillBgTilemapBufferRect(a, 0x8FD, b-2,   c+e, 1,   1, f);
+    FillBgTilemapBufferRect(a, 0x8FF, b-1,   c+e, 1,   1, f);
+    FillBgTilemapBufferRect(a, 0x900, b,     c+e, d-1, 1, f);
+    FillBgTilemapBufferRect(a, 0x901, b+d-1, c+e, 1,   1, f);
+    FillBgTilemapBufferRect(a, 0x902, b+d,   c+e, 1,   1, f);
```
Comment (Diegoisawesome) [inline src/main_menu.c]: Use the original value and BG_TILE_V_FLIP

File: src/main_menu.c [hunk 11]
```diff
@@ -1,15 +1,1955 @@
...
+    s16 *data = gTasks[taskId].data;
+    struct Sprite *sprite = &gSprites[gTasks[gUnknown_03000DD0].data[9]];
+
+    switch (data[0])
+    {
+        case 0:
+            if (sprite->callback == SpriteCallbackDummy)
+            {
+                sprite->oam.affineMode = 0;
+                goto _08030B98_inc_data0;
```
[the remark this answers is not in the document]
Comment (golemgalvanize) [reply src/main_menu.c]: Pikalax tried everything, this was a weird one dude

File: src/main_menu.c (hunk 7 above)
Comment (golemgalvanize) [reply src/main_menu.c]: Yes they are

File: src/main_menu.c [hunk 12]
```diff
@@ -1453,7 +1477,7 @@ void task_new_game_prof_birch_speech_part2_11(u8 taskId)
         spriteId = gTasks[taskId].data[2];
         gSprites[spriteId].callback = nullsub_11;
         SetGpuReg(REG_OFFSET_DISPCNT, DISPCNT_OBJ_ON | DISPCNT_OBJ_1D_MAP);
-        BeginNormalPaletteFade(-0x10000, 0, 0, 16, -1);
+        BeginNormalPaletteFade(-0x10000, 0, 0, 16, RGB_WHITEALPHA);
```
Comment (Diegoisawesome) [inline src/main_menu.c]: Oh also change -0x10000 to 0xFFFF0000

**Pull request merged**